src/clr-angular/data/datagrid/datagrid.ts

```diff
@@ -201,8 +241,45 @@ export class ClrDatagrid implements AfterContentInit, AfterViewInit, OnDestroy {
                 this.selectedChanged.emit(s);
             }
         }));
+        this.displayMode.view.subscribe(viewChange => {
+            while (this.projectedDisplayColumns.detach()) {
+            }
+            while (this.projectedCalculationColumns.detach()) {
+            }
+            while (this.calculationRows.detach()) {
+            }
+            while (this.displayedRows.detach()) {
+            }
+            if (viewChange === DatagridDisplayMode.DISPLAY) {
+                this.showDisplayTable = true;
+                this.renderer.removeClass(this.el.nativeElement, "datagrid-calculate-mode");
+                this.columns.forEach(column => {
+                    this.projectedDisplayColumns.insert(column.view);
+                });
+                this.rows.forEach(row => {
+                    this.displayedRows.insert(row.view);
+                });
+            } else if (viewChange === DatagridDisplayMode.CALCULATE) {
+                this.showDisplayTable = false;
+                this.renderer.addClass(this.el.nativeElement, "datagrid-calculate-mode");
+                this.columns.forEach(column => {
+                    this.projectedCalculationColumns.insert(column.view);
+                });
+                this.rows.forEach(row => {
+                    this.calculationRows.insert(row.view);
+                });
+            }
+        });
+        // Make sure columns get added when datagrid is first created.
+        this.columns.forEach(column => {
+            this.projectedDisplayColumns.insert(column.view);
+        });
+
+        // Moving here to try fixing conditional pagination issue.
     }
 
+    public display = true;
```

Comment:
This does not appear to be used anywhere.

Reply:
Will remove.